### PR TITLE
fix: Enhance datasource kernel_module_filters to check the loaded modules

### PR DESCRIPTION
**Pull request by wushiqinlou**
Signed-off-by: jiazhang <jiazhang@redhat.com>

### All Pull Requests:

Check all that apply:

* [x] Have you followed the guidelines in our Contributing document, including the instructions about commit messages?
* [ ] Is this PR to correct an issue?
* [x] Is this PR an enhancement?

### Complete Description of Additions/Changes:
The issue is https://github.com/RedHatInsights/insights-core/issues/3671

My first resolution is to add `keep_rc=True` into the spec:

~~~
modinfo_filtered_modules = command_with_args('modinfo %s', kernel_module_list.kernel_module_filters, `keep_rc=True`)
~~~
After updating, the spec can collect all of the output like this:

~~~
modinfo: ERROR: Module cvblk not found.
modinfo: ERROR: Module drbd not found.
modinfo: ERROR: Module emcpioc not found.
modinfo: ERROR: Module filepath not found.
filename:       /lib/modules/4.18.0-240.22.1.el8_3.x86_64/kernel/drivers/net/ethernet/intel/i40e/i40e.ko.xz
version:        2.8.20-k
...
~~~

However, the problem is that `modinfo: ERROR: Module XXX not found` can break the normal format:

~~~
alias:          pci:v00008086d000015ABsv*sd*bc*sc*i*
almodinfo: ERROR: Module krg_621845_imRH8_K2S_smp64 not found.
modinfo: ERROR: Module oracleafd not found.
ias:          pci:v00008086d000015B0sv*sd*bc*sc*i*
alias:          pci:v00008086d000015AAsv*sd*bc*sc*i*
...
name:           oracleasm
vermagic:       4.18.0-2modinfo: ERROR: Module oracleoks not found.
modinfo: ERROR: Module twnotify not found.
modinfo: ERROR: Module veeamsnap not found.
40.14.1.el8_3.x86_64 SMP mod_unload modversions
sig_id:         PKCS#7
~~~

We can modify the parser `KernelModulesInfo` to fix this issue, however, the broke format is not fixed, I am afraid that it is hard to avoid all the situations.

The second resolution is to update the datasource to only check the loaded modules:

~~~
cat insights/specs/datasources/kernel_module_list.py

@datasource(LsMod, HostContext)
def kernel_module_filters(broker):
    """
    Return a string of a list of modules from the spec filter,
    separated with space.
    """
    filters = sorted((get_filters(Specs.modinfo_modules)))
    if filters:
        new_modules = []
        for item in filters:
            if item in broker[LsMod]:
                new_modules.append(item)
        return ' '.join(new_modules)
    raise SkipComponent
~~~

### CODE REVIEW

File: insights/specs/datasources/kernel_module_list.py [hunk 1]
```diff
@@ -5,16 +5,21 @@
 from insights.core.exceptions import SkipComponent
 from insights.core.filters import get_filters
 from insights.core.plugins import datasource
+from insights.parsers.lsmod import LsMod
 from insights.specs import Specs
 
 
-@datasource(HostContext)
+@datasource(LsMod, HostContext)
```
Comment (xiangce) [inline insights/specs/datasources/kernel_module_list.py:12]: If a parser is required by a datasource, the data source should be `pre-loaded` while the collection, please add it to the "plugins.configs" section of "insights.collect.default_manifest" module, and set its `enable` as true.

File: insights/specs/datasources/kernel_module_list.py [hunk 2]
```diff
@@ -5,16 +5,21 @@
...
     separated with space.
     """
     filters = sorted((get_filters(Specs.modinfo_modules)))
     if filters:
-        return ' '.join(filters)
+        new_modules = []
+        for item in filters:
+            if item in broker[LsMod]:
+                new_modules.append(item)
+        return ' '.join(new_modules)
```
Comment (xiangce) [inline insights/specs/datasources/kernel_module_list.py:24]: Better raise a `SkipComponent` when no modules are found here.

**Review by xiangce — CHANGES_REQUESTED** [review]

File: insights/specs/datasources/kernel_module_list.py (hunk 2 above)
Comment (wushiqinlou) [reply insights/specs/datasources/kernel_module_list.py]: @xiangce - Updated, please check

File: insights/specs/datasources/kernel_module_list.py (hunk 1 above)
Comment (wushiqinlou) [reply insights/specs/datasources/kernel_module_list.py:12]: @xiangce - Updated, please check

**Review by xiangce — APPROVED** [review]

**Review by chenlizhong — APPROVED** [review]
Looks good to me

**Pull request merged**